controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -625,16 +625,16 @@ func (r *KubeadmControlPlaneReconciler) syncMachines(ctx context.Context, contro
 
 // reconcileControlPlaneConditions is responsible of reconciling conditions reporting the status of static pods and
 // the status of the etcd cluster.
-func (r *KubeadmControlPlaneReconciler) reconcileControlPlaneConditions(ctx context.Context, controlPlane *internal.ControlPlane) (ctrl.Result, error) {
```

Comment:
We usually keep the same signature for all the reconcile func

Reply:
But should we where those functions don't actually return anything? If so maybe we should have a `reconcileFunction` type to enforce this - like we do in cases where we use `reconcilePhases`

Reply:
I think if we shouldn't try to have similar function signatures across reconcilers if some of them don't use certain return parameters

Reply:
To add more context. I would prefer dropping unused parameters vs. keeping them to have similar functions with the same return parameters.

It just feels unnecessary to return and handle unused parameters and it makes the code easier to read if we only have and handle the parameters that we actually need.

Thought about it the other way around. If I would add a new reconcile func I would not add a not-used return paramter just to make the function more similar to others.